code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DefaultDeploymentUnitInstance.java

```diff
@@ -60,14 +56,6 @@ public void remove() {
                                 InstanceConstants.PROCESS_STOP, InstanceConstants.PROCESS_REMOVE));
             }
         }
-
-        // 2) remove the mapping
-        List<? extends ServiceExposeMap> maps = context.objectManager.mappedChildren(
```

Comment:
@sonchang  There can be only one map for instance->Service. I was just retrieving it not the most efficient way before. This map is always stored as a class member, and gets removed in the abstract class.